.github/workflows/sync_labels.yaml

```diff
@@ -0,0 +1,41 @@
+name: Sync labels
+
+on:
+  pull_request:
+    types: [opened, labeled, unlabeled]
+    branches: [master, next]
+  issues:
+    types: [labeled, unlabeled]
+
+jobs:
+  label-sync:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      issues: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history for all tags and branches
+
+      - name: Install dependencies
+        run: sudo apt-get install -y python3-github
+
+      - name: Pull request opened event
+        if: github.event.action == 'opened'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: python .github/scripts/sync_labels.py --repo ${{ github.repository }} --number ${{ github.event.number }} --action ${{ github.event.action }}
+
+      - name: Pull request labeled or unlabeled event
+        if: github.event_name == 'pull_request' && startsWith(github.event.label.name, 'backport')
```

Comment:
If the PR will open with a label - the workflow will run twice
1. Pull request opened event
2. Pull request labeled or unlabeled event

Correct?
If so, maybe adding another condition based on trigger-type

Reply:
Great,
Actually it's the same workflow run but matching two steps which is fine